[SITE] added block definition to the docs (#73)
blockprotocol/blockprotocol: site/src/_pages/docs/0_index.mdx

## Overview

The Block Protocol is a global standard for building interactive blocks connected to structured data.

The following pages provide a practical introduction for developers who want to:

- [develop blocks](https://blockprotocol.org/docs/developing-blocks) and publish them to the block registry
- [embed blocks](https://blockprotocol.org/docs/embedding-blocks) in their own application

For the full technical details, read [the Block Protocol specification](https://blockprotocol.org/spec).

### What's a block?

> Blocks are the basic units we use to build pages on the modern web. They are rectangular, full-width sections stacked on top of one another – also known as "block-level" elements.

Author: Updated and removed this language 👍

> 

Reviewer: ```suggestion

A block is any valid HTML published to the [Block Hub](https://blockprotocol.org/hub).

```

cc @CiaranMn @MaggieAppleton per our 'quickest start guide' @ https://blockprotocol.org/docs/developing-blocks

Author: Rephrased this but tried to keep the same gist in the "What is a Block Protocol block" section

> Blocks can be created using plain HTML, or by using a block editor interface like Notion or Wordpress' Gutenberg. With block editors you can usually rearrange the order of blocks on a page using drag and drop.

Reviewer: ```suggestion
Blocks can be inserted using plain HTML/JavaScript, or through visual interfaces provided by modern content management systems and applications like all-in-one workspaces. With these block-based editors you can usually rearrange the order of blocks on a page using drag and drop.
```

Author: Updated this to read:

> Blocks can be added to webpages using plain HTML and/or JavaScript. Some modern content management systems (eg. Wordpress, Squarespace) and applications (eg. Notion, Coda) allow you to add blocks to a page using a visual interface. With these block-based editors you can usually rearrange the order of blocks on a page using drag and drop. Blocks can be stacked on top of one another, arranged in columns, or embedded inside other blocks.

I know pointing to specific apps isn't ideal, so could remove the examples here. It would help paint a more concrete picture in people's minds of what we're talking about though.

> Each block has a type that determines how it displays data. You're probably familiar with the most common block types such as _paragraph_, _header_, _image_, _list_, and _table_. Some block types have more advanced interactive features such as filtering, search, and sorting.

Reviewer: cc @CiaranMn for feedback

> There are an inifinite number of possible block types. One goal of the Block Protocol is to encourage developers to build and share brand new block types we've never seen before.

Reviewer: ```suggestion
There are an infinite number of possible block types. One goal of the Block Protocol is to encourage developers to build and share brand new types of blocks that haven't been seen before. We don't believe application's users should be limited to a dozen or so blocks (the bounded imaginations or expected maintenance bandwidth of their developers).
```

Should be rephrased to make simpler, but adds a bit of missing context re: the why. Can be improved now or later.

Author: Updated with suggestions 👍

> From a developer perspective, blocks are similar to [web components](https://en.wikipedia.org/wiki/Web_Components). They are reusable sets of HTML elements that can accept any data that matches the block's schema. For example, an image block can only accept an image URL or file, but not a text string.

Reviewer 2: I'm assuming this is talking about generic blocks, rather than BP blocks - we could add the sentence I suggest above after this ("A block which complies..."). 

This could be rephrased but I'm trying to get across that:
- web components are an implementation method rather than an alternative to blocks
- "schema" is something we use rather than being universally in use for different versions of "blocks"
- pedantry: tweaked the example to make it starker, as technically an image URL is a text string
 
```suggestion
From a developer perspective, blocks offer encapsulation of specific functionality ([web components](https://en.wikipedia.org/wiki/Web_Components) are one possible way of implementing a block). They are reusable elements that expect certain data to be passed to them (what we refer to as a block's "schema"). For example, an image block might only accept an image URL in its `src` property, not a number.
```

Reviewer 2: I'd also move the "goals of the BP" to the bottom as I think it'd flow better with a new para on BP blocks. So I'm suggesting the final paras would be something like this. Maybe with the paras swapped?

> There are an infinite number of possible block types. One goal of the Block Protocol is to encourage developers to build and share brand new types of blocks that haven't been seen before, and for applications to offer users far more blocks than any one team could imagine or build.
> 
> A block which complies with the Block Protocol has the added advantage of specifying the data it receives - and modifying it - in a standardized way. This makes it easier to re-use Block Protocol blocks across applications - and if they are published to the Block Hub, easily discoverable by users.

Each block has a type that determines the kind of data it can display and edit. You're probably familiar with the most common block types such as _paragraph_, _header_, _image_, _list_, and _table_. Some block types have more advanced interactive features such as filtering, search, and sorting.

There are an infinite number of possible block types. One goal of the Block Protocol is to encourage developers to build and share brand new types of blocks that haven't been seen before, and for applications to offer users far more blocks than any one team could imagine or build.

Blocks can be added to webpages using plain HTML and/or JavaScript. Some modern content management systems (eg. Wordpress, Squarespace) and applications (eg. Notion, Coda) allow you to add blocks to a page using a visual interface. With these block-based editors you can usually rearrange the order of blocks on a page using drag and drop. Blocks can be stacked on top of one another, arranged in columns, or embedded inside other blocks.

> From a developer perspective, blocks encapsulate a set of specific functionality. Building a block is no different to writing a React, Vue, or [Web Component](https://en.wikipedia.org/wiki/Web_Components) – in fact, these are all ways you might implement a block. They are reusable elements that expect certain data to be passed to them (what we refer to as a block's “schema”). For example, an image block might only accept an image URL in its `src` property, not a number.

Reviewer 3: ```suggestion
From a developer perspective, blocks encapsulate a set of specific functionality. Building a block is no different to writing a React, Vue, or [Web Component](https://en.wikipedia.org/wiki/Web_Components) – in fact, these are all ways you might implement a block. They are reusable elements that expect certain data to be passed to them (what we refer to as a block's ‘schema’). For example, an image block might only accept an image URL in its `src` property, not a number.
```

The most common rule I‘ve seen was: _“” for quotes, ‘’ for emphasis_. We don’t have to stick with it rigorously, but it might be worth keeping the rule in our heads.

For the full technical details, read [the Block Protocol specification](https://blockprotocol.org/spec).

### Why is structured data useful?

> Structured data allows computers to understand information. It is easier to create and easier to understand than unstructured data.

Reviewer: "It is easier to create and easier to understand than unstructured data." I take issue with the claim that it's easier to create. The Block Protocol was (partly) born of frustration that this wasn't the case. When in rapid-notetaking mode most people just want to dump ideas onto a page and not worry about structure. But blocks that are both useful to folks in that type of a setting (where speed matters), and which organically add semantic meaning to the information being captured at the same time, lay the groundwork for this to be true.

Author: I was unsure about that claim too - copied off the rough notes from Joel, but figured it was in the context of pre-selecting entities from a known list vs. relying on memory. We can adapt to "Structured data makes it much easier for computers to understand information"

Author: Updated to: "Structured data makes it much easier for computers to understand the information we give them."

> Most of the data on the web is unstructured. When we write webpages in HTML, it's easily readable by humans, but not by computers. The computer only knows you've added a bunch of `<p>` tags to a page, but doesn't know anything about what you've written inside them.

Reviewer: ```suggestion
Most of the data on the web is unstructured. When we write webpages in HTML, it's easily readable by humans, but not by computers. The computer only knows you've added a bunch of `<p>` tags to a page, but doesn't understand the contents or shape of what you've written inside (at least without any fancy natural language processing).
```

> Unlike plain HTML, blocks created with the Block Protocol have **types**, such as _Person_, _Company_, and _Date_ that help computers understand what your text is about.

Reviewer: ```suggestion
Unlike plain HTML, blocks created with the Block Protocol have **types**, such as _Person_, _Company_, and _Date_ that help computers understand what they actually contain.
```

> {diagram of HTML vs. structure block data}

Reviewer: @MaggieAppleton We should ready this for close of play Wednesday or remove this placeholder from the docs. Do you think we'll have something for then?

Imagine you are creating a webpage to announce a book signing event with a well-known author. You could add an _Author_ block, a _Location_ block, and a _Time_ block to this page. Instead of being some mystery HTML, your information is now readable by computers. It would be much easier to automatically pass this information to other websites and applications, such as a calendar application, or the author's personal website.

> We are trying to fulfil the vision of the Semantic Web – the original Web 3.0 (before all the NFTs took over). Our goal is to update the web from a pile of randomly formatted, human-readable pages into a structure trove of knowledge.

Reviewer: ```suggestion
We are trying to fulfil the vision of the Semantic Web (aka. the OG Web3, before all the NFTs took over!), transforming the web's randomly formatted, human-readable pages into structured troves of connected knowledge, improving both user and developer experiences as we go.
```
